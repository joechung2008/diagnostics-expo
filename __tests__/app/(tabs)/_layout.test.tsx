import TabsLayout from '@/app/(tabs)/_layout';
import { render } from '@testing-library/react-native';
import React from 'react';

// Mock ThemeContext
jest.mock('@/components/ThemeContext', () => ({
  useTheme: () => ({
    colors: {
      background: '#ffffff',
      surface: '#f5f5f5',
      surfaceSecondary: '#e0e0e0',
      text: '#000000',
      border: '#cccccc',
      primary: '#007AFF',
      textSecondary: '#8E8E93',
    },
  }),
}));

// Mock Header component
jest.mock('@/components/Header', () => ({
  __esModule: true,
  default: jest.fn(() => null),
}));

// Mock TabsContainer component
jest.mock('@/components/TabsContainer', () => ({
  __esModule: true,
  default: jest.fn(() => null),
}));

// Mock SafeAreaView
jest.mock('react-native-safe-area-context', () => ({
  SafeAreaView: ({ children, style }: any) => {
    const { createElement } = jest.requireActual('react');
    return createElement('SafeAreaView', { style }, children);
  },
}));

describe('TabsLayout', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders with correct structure and theme colors', () => {
    const { toJSON } = render(<TabsLayout />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders SafeAreaView with correct styles', () => {
    const { toJSON } = render(<TabsLayout />);
    // Since we're using snapshot testing, we rely on the snapshot to verify styles
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders Header component', () => {
    const mockHeader = jest.requireMock('@/components/Header').default;
    render(<TabsLayout />);
    expect(mockHeader).toHaveBeenCalledTimes(1);
  });

  it('renders TabsContainer component', () => {
    const mockTabsContainer = jest.requireMock(
      '@/components/TabsContainer'
    ).default;
    render(<TabsLayout />);
    expect(mockTabsContainer).toHaveBeenCalledTimes(1);
  });

  it('applies theme background color to container', () => {
    const { toJSON } = render(<TabsLayout />);
    // The snapshot will capture the background color from theme
    expect(toJSON()).toMatchSnapshot();
  });

  it('uses flex: 1 container style', () => {
    const { toJSON } = render(<TabsLayout />);
    // The snapshot will capture the flex: 1 style
    expect(toJSON()).toMatchSnapshot();
  });
});
