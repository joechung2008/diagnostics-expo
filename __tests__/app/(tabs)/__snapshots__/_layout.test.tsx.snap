// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsLayout applies theme background color to container 1`] = `
<SafeAreaView
  style={
    [
      {
        "flex": 1,
      },
      {
        "backgroundColor": "#ffffff",
      },
    ]
  }
/>
`;

exports[`TabsLayout renders SafeAreaView with correct styles 1`] = `
<SafeAreaView
  style={
    [
      {
        "flex": 1,
      },
      {
        "backgroundColor": "#ffffff",
      },
    ]
  }
/>
`;

exports[`TabsLayout renders with correct structure and theme colors 1`] = `
<SafeAreaView
  style={
    [
      {
        "flex": 1,
      },
      {
        "backgroundColor": "#ffffff",
      },
    ]
  }
/>
`;

exports[`TabsLayout uses flex: 1 container style 1`] = `
<SafeAreaView
  style={
    [
      {
        "flex": 1,
      },
      {
        "backgroundColor": "#ffffff",
      },
    ]
  }
/>
`;
